src/app/pages/product/product-page.component.ts

```diff
@@ -142,7 +149,9 @@ export class ProductPageComponent implements OnInit, OnDestroy {
       )
       .subscribe(([product, category]) => {
         this.ngZone.run(() => {
-          this.router.navigateByUrl(generateProductUrl(product, category), { replaceUrl });
+          this.router.navigateByUrl('/', { skipLocationChange: true }).then(() => {
+            this.router.navigateByUrl(generateProductUrl(product, category), { replaceUrl });
+          });
```

Comment:
For my understanding: Why is this necessary? What does this fix?

Reply:
in case of redirect to itself (found by findPossibleVariationForSelection), the page lifecycle does not work. (corner case)

Reply:
Could you try activating this? https://angular.io/api/router/ExtraOptions#onSameUrlNavigation

Reply:
i tryed onSameUrlNavigation, but it doesnt work
https://drive.google.com/file/d/1TKxCFneonFgCR12o7SAJeZ6PxBEVKxhW/view